include/ignition/gui/GuiEvents.hh

```diff
@@ -43,6 +43,158 @@ namespace ignition
         /// \brief Unique type for this event.
         static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser);
       };
+
+      /// \brief The class for sending and receiving custom snap value events.
+      class SnapIntervals : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _xyz XYZ snapping values.
+        /// \param[in] _rpy RPY snapping values.
+        /// \param[in] _scale Scale snapping values.
+        public: SnapIntervals(
+                    const math::Vector3d &_xyz,
+                    const math::Vector3d &_rpy,
+                    const math::Vector3d &_scale)
+        : QEvent(kType), xyz(_xyz), rpy(_rpy), scale(_scale)
+        {
+        }
+  
+        /// \brief Get the XYZ snapping values.
+        /// \return The XYZ snapping values.
+        public: math::Vector3d XYZ() const
+        {
+          return this->xyz;
+        }
+  
+        /// \brief Get the RPY snapping values.
+        /// \return The RPY snapping values.
+        public: math::Vector3d RPY() const
+        {
+          return this->rpy;
+        }
+  
+        /// \brief Get the scale snapping values.
+        /// \return The scale snapping values.
+        public: math::Vector3d Scale() const
+        {
+          return this->scale;
+        }
+  
+        /// \brief The QEvent representing a snap event occurrence.
+        static const QEvent::Type kType = QEvent::Type(QEvent::User);
```

Comment:
be5a482d8da3da972dff28a187723c805378d367

Reply:
I updated the events to have different ID's.  I was thinking we could also have them share the same ID's and then try to cast as both the `ignition::gui` and `ignition::gazebo::gui` events if that type is found, since they technically represent the same event as well.  Not sure which approach you think would be better.  On one hand we have two separate events for the same functionality, on the other hand, we have to add a little more redundancy into the event casting.

Reply:
Oh interesting idea with the casting. One thing I think could be risky is trying to keep both in sync though... In the past we've tried to keep enums in sync across libraries and it can become a headache :confused: I recognize that the situation here would just be a temporary transition though.